code/game/objects/effects/spawners/random.dm

```diff
@@ -566,3 +566,113 @@
 					/obj/item/ammo_magazine/pistol/derringer,\
 					/obj/item/ammo_magazine/revolver/cmb,\
 					/obj/item/ammo_magazine/pistol/standard_pocketpistol)
+
+
+////new test
+
+/obj/effect/spawner/random_set
+	name = "Random Object"
+	desc = "This item type is used to spawn random objects at round-start"
+	icon = 'icons/effects/landmarks_static.dmi'
+	icon_state = "random_loot"
+	/// this variable determines the likelyhood that this random object will not spawn anything
+	var/spawn_nothing_percentage = 0
+	///the list of what actually gets spawned
+	var/list/spawned_gear_list
+	///the list of lists, providing the options that can be spawned
```

Comment:
you should probably add an extra comment on how exactly a mapper should format and what the nested list does

Reply:
done